src/main/java/hudson/plugins/git/extensions/impl/SubmoduleOption.java

```diff
@@ -42,15 +42,17 @@
     private String reference;
     private boolean parentCredentials;
     private Integer timeout;
+    private Integer threads;
```

Comment:
I'm a bit confused.  I don't see you using `java.lang.Thread`.  I only see this `Integer` which I assume is meant for a thread count.  I also don't see any usage of thread methods like joining or locking.  Can you explain to me how this is adding parallel submodule updating?

Reply:
I agree that it would be nice to match the number of cores, but I'm not sure this is very practical. Even if there is an API call to get the number of cores (I have no idea if there is), it would only make sense if the build is executed on the Jenkins master. If the build is run on a remote node, then the user would obviously want to set this value based on the core count there.

I think that 1 is still a sensible default, though I would agree that 8 is a sensible maximum value, since anything more than that is likely to cause problems elsewhere. I'll add a fixup to add this.

Reply:
I also think that 1 should be the default.
The upper bound shouldn't be limited and completely be up to the user.
@nre-ableton maybe you misunderstood @samrocketman when he was talking about the maximum value. He talked about limiting the value when the smart default is not 1 but detected in a smart way.

Reply:
@darxriggs Yes, sorry, I think I did misunderstand. Thanks for the clarification. I'll remove my fixup.

Reply:
Yes, I meant we did not want to auto-detect abusing a user's system so limiting it without setting too high of a default.  I think we're all on the same page, though :smile: